Convert nvcomp-feedstock to Python API feedstock

carterbox opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This feedstock now ships the python API of nvcomp by revending the python modules from PYPI wheels. The C API is stripped from the wheels, and instead we depend on `libnvcomp-dev` from the new libnvcomp-feedstock. The libnvcomp-feedstock repackages the binaries and cmake files from the redist tarball. 

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Closes #22
Closes #24
Closes #21
Closes #20

carterbox committed:
BLD: Replace C-API package with python-API

carterbox committed:
BLD: Cross-compile for linux

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ No valid build backend found for Python recipe for package `nvcomp` using `pip`. Python recipes using `pip` need to explicitly specify a build backend in the `host` section. If your recipe has built with only `pip` in the `host` section in the past, you likely should add `setuptools` to the `host` section of your recipe.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/16381732656](https://github.com/conda-forge/conda-forge-webservices/actions/runs/16381732656). Examine the logs at this URL for more detail.</sub>

regro-cf-autotick-bot committed:
Upgrade to CUDA 12.9

CUDA 12.8 added support for architectures `sm_100`, `sm_101` and `sm_120`,
while CUDA 12.9 further added `sm_103` and `sm_121`. To build for these,
maintainers will need to modify their existing list of specified architectures
(e.g. `CMAKE_CUDA_ARCHITECTURES`, `TORCH_CUDA_ARCH_LIST`, etc.)
for their package. A good balance between broad support and storage
footprint (resp. compilation time) is to add `sm_100` and `sm_120`.

Since CUDA 12.8, the conda-forge nvcc package now sets `CUDAARCHS` and
`TORCH_CUDA_ARCH_LIST` in its activation script to a string containing all
of the supported real architectures plus the virtual architecture of the
latest. Recipes for packages who use these variables to control their build
but do not want to build for all supported architectures will need to override
these variables in their build script.

ref: https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#new-features

carterbox committed:
MNT: Re-rendered with conda-build 25.5.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.07.18.21.26.56

billysuh7 commented on recipe/meta.yaml:
```diff
@@ -1,91 +1,54 @@
-    - arm-variant * {{ arm_variant_type }}  # [aarch64]
-    - cf-nvidia-tools 1  # [linux]
+    - cf-nvidia-tools 1.*  # [linux]
+    - cross-python_{{ target_platform }}    # [build_platform != target_platform]
```
I am curious where `cross-python_{{ target_platform }}` comes into play?

carterbox replied:
This recipe is using the cross-target conda-forge infrastructure to create the aarch64 package. It doesn't matter that we aren't actually compiling anything. We have to pretend that we are actually building from source because there could be side effects from the cross-python package.

carterbox committed:
REF: Rename python package from nvcomp-python to just nvcomp

carterbox committed:
DOC: Add cudart license back to recipe

carterbox commented:
This PR is ready, but it's Friday, and we should wait until RAPIDS is ready. @bdice, could you please post here or merge once RAPIDS has migrated/repodata-patched it's packages to use libnvcomp where appropriate? I can also make backports of libnvcomp if you want older versions of that package available.

bdice commented:
Here are PRs to RAPIDS packages that use `libnvcomp`. Once these are in, this change can be merged.
- https://github.com/rapidsai/kvikio/pull/774
- https://github.com/rapidsai/cudf/pull/19439

bdice commented:
@carterbox The RAPIDS changes are now merged. It should be fine to update `nvcomp` to be the Python library now.
